[BUG] Spark 411 unit test fails with NoSuchMethodError RowDeltaUtils.REINSERT_OPERATION in RapidsShuffleIntegrationSuite

**Describe the bug**
Build: rapids_scala213_nightly-dev-github/725

The Spark 411 unit test against cuda12 failed in the RapidsShuffleIntegrationSuite with a NoSuchMethodError for RowDeltaUtils.REINSERT_OPERATION. This indicates a binary incompatibility between the compiled 411 shim and the Spark 4.1.1 runtime.

Error logs:
```
mvn -Dbuildver=411 -f scala2.13/ -B -Dmaven.wagon.http.retryHandler.count=5 -s \
jenkins/settings.xml -P mirror-apache-to-urm clean package -Dcuda.version=cuda12 \
-Drat.skip=true -Dmaven.javadoc.skip=true -Dmaven.scalastyle.skip=true

...
RapidsShuffleIntegrationSuite:
*** RUN ABORTED ***
  java.lang.NoSuchMethodError: 'int org.apache.spark.sql.catalyst.util.RowDeltaUtils$.REINSERT_OPERATION()'
  at org.apache.spark.sql.catalyst.analysis.RewriteRowLevelCommand.$init$(RewriteRowLevelCommand.scala:42)
  at org.apache.spark.sql.catalyst.analysis.RewriteDeleteFromTable$.<clinit>(RewriteDeleteFromTable.scala:38)
  at org.apache.spark.sql.catalyst.analysis.Analyzer.batches(Analyzer.scala:497)
  at org.apache.spark.sql.catalyst.rules.RuleExecutor.execute(RuleExecutor.scala:237)
  at org.apache.spark.sql.catalyst.analysis.Analyzer.org$apache$spark$sql$catalyst$analysis$Analyzer$$executeSameContext(Analyzer.scala:343)
  at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$execute$1(Analyzer.scala:339)
  at org.apache.spark.sql.catalyst.analysis.AnalysisContext$.withNewAnalysisContext(Analyzer.scala:224)
  at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:339)
  at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:289)
  at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$executeAndTrack$1(RuleExecutor.scala:207)

[INFO] RAPIDS Accelerator for Apache Spark Tests .......... FAILURE [01:34 min]
[ERROR] Failed to execute goal org.scalatest:scalatest-maven-plugin:2.0.2:test (test) on project rapids-4-spark-tests_2.13: There are test failures
```

**Environment details**
- Scala 2.13, Spark 4.1.1 (buildver=411), CUDA 12
- Test suite: RapidsShuffleIntegrationSuite in rapids-4-spark-tests_2.13
- Likely caused by Spark 4.1.1 API change in RowDeltaUtils (method signature/visibility changed between snapshots); the 411 shim needs updating to match.

--- a/pom.xml
+++ b/pom.xml
@@ -710,6 +710,8 @@
                 <spark.test.version>${spark411.version}</spark.test.version>
                 <parquet.hadoop.version>1.13.1</parquet.hadoop.version>
                 <rapids.delta.artifactId1>rapids-4-spark-delta-stub</rapids.delta.artifactId1>
+                <iceberg.artifact.suffix>${spark40x.iceberg.artifact.suffix}</iceberg.artifact.suffix>
+                <iceberg.runtime.version>${iceberg.110x.version}</iceberg.runtime.version>
                 <slf4j.version>2.0.7</slf4j.version>
                 <scala.javac.args>-Xlint:all,-serial,-path,-try,-processing,-options|-Werror</scala.javac.args>
             </properties>

--- a/scala2.13/pom.xml
+++ b/scala2.13/pom.xml
@@ -710,6 +710,8 @@
                 <spark.test.version>${spark411.version}</spark.test.version>
                 <parquet.hadoop.version>1.13.1</parquet.hadoop.version>
                 <rapids.delta.artifactId1>rapids-4-spark-delta-stub</rapids.delta.artifactId1>
+                <iceberg.artifact.suffix>${spark40x.iceberg.artifact.suffix}</iceberg.artifact.suffix>
+                <iceberg.runtime.version>${iceberg.110x.version}</iceberg.runtime.version>
                 <slf4j.version>2.0.7</slf4j.version>
                 <scala.javac.args>-Xlint:all,-serial,-path,-try,-processing,-options|-Werror</scala.javac.args>
             </properties>

